lib/beaker/options/parser.rb

```diff
@@ -228,10 +228,22 @@ def parse_hosts_options
         hosts_options = Beaker::Options::HostsFileParser.parse_hosts_file(@options[:hosts_file])
         return hosts_options
       rescue Errno::ENOENT
-        require 'tempfile'
+        error_message = "\nHosts file '#{@options[:hosts_file]}' does not exist."
+        error_message << "\nTrying as beaker-hostgenerator input.\n\n"
+        $stderr.puts error_message
         require 'beaker-hostgenerator'
-        bhg_cli = BeakerHostGenerator::CLI.new( [ @options[:hosts_file] ] )
-        hosts_file_content = bhg_cli.execute
+
+        hosts_file_content = begin
+          bhg_cli = BeakerHostGenerator::CLI.new( [ @options[:hosts_file] ] )
+          bhg_cli.execute
+        rescue BeakerHostGenerator::Exceptions::Error,
+          BeakerHostGenerator::Exceptions::InvalidNodeSpecError => error
+          error_message = "\nbeaker-hostgenerator was not able to use this value as input."
+          error_message << "\nExiting with an Error.\n\n"
+          $stderr.puts error_message
+          raise error
```

Comment:
Looked into the [dependencies between these two](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/cli.rb#L14-L16). Logger directly depends on the options to determine how it should be setup.

The option that comes to mind would be to create a logger specifically for the phase before the "real logger" is configured post options parsing. However, that doesn't seem like a good option to me, as it's an unconfigurable logger. At least this way, the user has the standard control mechanisms for the basic OS output streams.

For your comment about both printing to standard error and raising an exception, I decided to have it that way to give the most information possible. I'm not familiar with anyone using beaker as a library, in my experience it's just an application. The printing is to help someone get the context that they need to be able to deal with the error message. This is particularly helpful if they weren't intending to send the string to beaker-hostgenerator, as I've seen happen already with people using the master branch.